src/hdf5/cloudsat_netcdf_to_ioda.py

```diff
@@ -0,0 +1,219 @@
+    # in addition to the ones already loaded in from the input file
+    dtg = False  # pass a reference time for the observation such as to an NWP analysis
+    if dtg:
+        GlobalAttrs['datetimeReference'] = dtg.strftime("%Y-%m-%dT%H:%M:%SZ")
+    GlobalAttrs['converter'] = os.path.basename(__file__)
+
+    # set key for observable
+    k = radarReflectivityAtt
+
+    # pass parameters to the IODA writer
+    VarDims = {
+        k: ['Location', 'Channel'],
+        'sensorChannelNumber': ['Channel'],
+    }
+
+    DimDict = {
+        'Location': nlocs,
+        'Channel': obs_data[('sensorChannelNumber', metaDataName)],
+    }
+    writer = iconv.IodaWriter(output_filename, locationKeyList, DimDict)
+
+    VarAttrs = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
+    set_obspace_attributes(VarAttrs)
+    set_metadata_attributes(VarAttrs)
+
+    VarAttrs[(k, 'ObsValue')]['_FillValue'] = float_missing_value
+    VarAttrs[(k, 'ObsError')]['_FillValue'] = float_missing_value
+    VarAttrs[(k, 'PreQC')]['_FillValue'] = int_missing_value
+    VarAttrs[(k, 'ObsValue')]['units'] = 'dBz'
+    VarAttrs[(k, 'ObsError')]['units'] = 'dBz'
+
+    # final write to IODA file
+    writer.BuildIoda(obs_data, VarDims, VarAttrs, GlobalAttrs)
+
+
+def import_obs_data(cpr_obs):
+
+    # appears to be observation cleansing and conditioning
+    cpr_obs = cpr_obs.rename_vars({"elevation": "elevation1"})
+    cpr_obs = cpr_obs.stack(Location=['obs_id', 'elevation']).reset_index("Location")
+    cpr_obs = cpr_obs.transpose("Location", "channel")
+
+    locid = (cpr_obs.obs.values < -100) | (cpr_obs.obs.values > 100) | np.isnan(cpr_obs.obs.values) | np.isinf(cpr_obs.obs.values)
+    locid = cpr_obs.Location.values[np.sum(locid, axis=1) == 0]
+    cpr_obs = cpr_obs.isel(Location=locid)
+    # end conditioning and cleansing block
+
+    # this function will map the cloud radar data in the cpr xarray
+    # into a dictionary to be passed to the IODA writer functions
+    nobs = cpr_obs.Location.size
+    nchans = cpr_obs.channel.size
+
+    # ideally an attribute from the file read in
+    # example: cpr_obs.attrs['ShortName'].decode("utf-8")
+    # here we are hardcoding to the function the name of the satellite
+    WMO_sat_ID = get_WMO_satellite_ID('CloudSat')
+
+    # allocate space for output depending on which variables are to be saved
+    obs_data = init_obs_loc()
+
+    obs_data[('latitude', metaDataName)] = cpr_obs.lat.values.astype(np.float32)
+    obs_data[('longitude', metaDataName)] = cpr_obs.lon.values.astype(np.float32)
+    obs_data[('satelliteIdentifier', metaDataName)] = np.full((nobs), WMO_sat_ID, dtype='int32')
+    obs_data[('sensorCentralFrequency', metaDataName)] = np.array([94.05]).astype(np.float32)    # cpr_sim.Frequency.values.astype(np.float32)
+    obs_data[('sensorCentralWavenumber', metaDataName)] = np.array([3.1371]).astype(np.float32)  # cpr_sim.Wavenumber.values.astype(np.float32)
```

Comment:
@imoradi these too assume the RADAR has a single channel and the frequency is fixed ?